actions/release-candidate/dist/main/index.js

```diff
@@ -31855,46 +31855,59 @@ async function run() {
 		const tlp_dir = core.getInput("tlp_dir", { required: true });
 		const project_id = core.getInput("project_id", { required: true });
 		const project_dir = core.getInput("project_dir");
-		const gpg_signing_key = core.getInput("gpg_signing_key", { required: true });
-		const svn_username = core.getInput("svn_username", { required: true });
-		const svn_password = core.getInput("svn_password", { required: true });
-		const nexus_username = core.getInput("nexus_username", { required: true });
-		const nexus_password = core.getInput("nexus_password", { required: true });
-		let publish = core.getBooleanInput("publish");
-
-		// import signing key into gpg and get it's key id
-		let gpg_import_stdout = ""
-		await exec("gpg", ["--batch", "--import", "--import-options", "import-show"], {
-			input: Buffer.from(gpg_signing_key),
-			listeners: {
-				stdout: (data) => { gpg_import_stdout += data.toString(); }
-			}
-		});
-		const gpg_signing_key_id = gpg_import_stdout.match("[0-9A-Z]{40}")[0];
-		console.info("Using gpgp key id: " + gpg_signing_key_id);
+		const publish = core.getBooleanInput("publish");
+
+		// get the actual project version, this requires a 'VERSION' file at
+		// the root of the repository
+		const project_version = fs.readFileSync("VERSION").toString().trim();
+		const is_snapshot = project_version.includes("-SNAPSHOT");
+		const is_apache = process.env.GITHUB_REPOSITORY_OWNER == "apache";
+		const gitTagPrefix = "refs/tags/";
+		const is_tagged = github.context.eventName == "push" && github.context.ref.startsWith(gitTagPrefix);
+		const do_publish =
+			// Note that publishing could be disabled if the publish input was explicitly set
+			// to false
+			publish
+			// require a pushed tag to enable publishing
+			&& is_tagged
+			// require non-snapshot ASF builds to enable publishing
+			&& (!is_snapshot && is_apache)
+		const gpg_signing_key = core.getInput("gpg_signing_key", {required: do_publish});
+
+		let gpg_signing_key_id = "";
+		if (gpg_signing_key.trim() === "") {
+			// Generate keypair (non-interactive)
+			await exec("gpg", ["--batch", "--yes", "--passphrase", '', "--quick-generate-key", process.env.USER ], {
+				silent: true
+			});
+		} else {
+			// import signing key into gpg
+			await exec("gpg", ["--batch", "--import", "--import-options", "import-show"], {
+				input: Buffer.from(gpg_signing_key)
+			});
+		}
 
-		// tags must be signed with a committers key, download and import committer
-		// keys for verification later
-		let committer_keys = "";
-		await exec("curl", [`https://downloads.apache.org/${ tlp_dir }/KEYS`], {
+		// Capture the key id of the most recent generated/imported key
+		let gpg_list_secret_keys_stdout = "";
+		await exec("gpg", ["--list-secret-keys", "--with-colons"], {
 			silent: true,
 			listeners: {
-				stdout: (data) => { committer_keys += data.toString(); }
+				stdout: (data) => {
+					gpg_list_secret_keys_stdout += data.toString().trim()
```

Comment:
I think trim() removes newlines, so if --list-secret-keys output multiple lines they would all get squashed to a single line and break the split('\n') below. Can we can just do `... += data.toString()` just in case? I imagine the .split.findLast stuff should still work without trimmed content?